orchagent/aclorch.cpp

```diff
@@ -2097,10 +2153,17 @@ bool AclTable::addStageMandatoryMatchFields()
             // Add the stage particular matching fields
             for (auto match : fields_for_stage[stage])
             {
-                type.addMatch(make_shared<AclTableMatch>(match));
-                SWSS_LOG_INFO("Added mandatory match field %s for table type %s stage %d",
-                                sai_serialize_enum(match, &sai_metadata_enum_sai_acl_table_attr_t).c_str(),
-                                type.getName().c_str(), stage);
+                if (match != SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE)
+                {
+                    type.addMatch(make_shared<AclTableMatch>(match));
+                    SWSS_LOG_INFO("Added mandatory match field %s for table type %s stage %d",
+                        sai_serialize_enum(match, &sai_metadata_enum_sai_acl_table_attr_t).c_str(),
+                        type.getName().c_str(), stage);
+                }
+                else
+                {
+                    addStageMandatoryRangeFields();
```

Comment:
Is it possible that `addStageMandatoryRangeFields` is called for multiple times when walking through the for loop?